### DISCUSSION TITLE
Is it possible to move the entry module?

### DISCUSSION THREAD

**Discussion by sazzer** [wailsapp/wails#1955] [Q&A]
I've just come across and started using Wails and it looks great :)
I'm curious though if it's possible to move the module that represents the main application so it's not in the root directory? It's just a cleanliness thing really - the frontend has all of the code under `src`, but in the backend it seems that whilst I can put the majority of the code wherever I want, the actual main module must be the root of the project.

Literally I'm just wondering about putting `main.go` in a separate directory instead - e.g. `cmd/wails` for example.

Cheers

**Comment by alanchenchen** [comment]
Absolutely possible. But it will be much more complex than wails-cli for no-gopher.
Mannual build is helpful for you https://wails.io/docs/guides/manual-builds/.
Using go compiler `go build cmd/wails` could compile sub directory go module but entry module named `main` is necessary.

**Comment by mholt** [comment]
I think the latest commits will do what you want!

https://github.com/wailsapp/wails/pull/2009
